elastic/kibana
# [Search][Dashboard] Restore searchSessionId from URL

Build on top of https://github.com/elastic/kibana/pull/81297
Part of https://github.com/elastic/kibana/issues/61738

## Summary

* Restore `searchSessionId` from URL
* Add `searchSessionId` support to dashboard's URL generator

### Additional info

* Didn't do any additional handling to return to background search using back button. I spent some time trying to make it reliable, but I think to make it work in 100% of case it will require additional work and refactoring around dashboard state management. (Unless I missed something obvious)  Created a ticket, please see explanation of the problem: https://github.com/elastic/kibana/issues/82419

### How to test 

You can build a dashboard URL with fake session id: 

```
${dashboardURL}&searchSessionId=__fake__
```

Open it.
And you should see `__fake__` as session id in inspector.

^^ this is what a functional test does


### Checklist

Delete any items that are not applicable to this PR.
